app/controllers/registrations_controller.rb

```diff
@@ -2,56 +2,55 @@ class RegistrationsController < Devise::RegistrationsController
   prepend_before_action :require_no_authentication, only: []
 
   def new
-    if user_signed_in?
-      redirect_to root_path(signin: "true")
-    else
-      if URI(request.referer || "").host == URI(request.base_url).host
-        store_location_for(:user, request.referer)
-      end
-      super
+    return redirect_to root_path(signin: "true") if user_signed_in?
+
+    if URI(request.referer || "").host == URI(request.base_url).host
+      store_location_for(:user, request.referer)
     end
+
+    super
   end
 
-  # rubocop:disable Metrics/CyclomaticComplexity
-  # rubocop:disable Metrics/PerceivedComplexity
   def create
-    not_authorized unless Settings::Authentication.allow_email_password_registration ||
-      Settings::General.waiting_on_first_user
-    not_authorized if Settings::General.waiting_on_first_user && ENV["FOREM_OWNER_SECRET"].present? &&
-      ENV["FOREM_OWNER_SECRET"] != params[:user][:forem_owner_secret]
+    not_authorized unless email_registrable?
```

Comment:
Since we are returning a `not_authorized` here may this be a hint that we should add a `RegistrationPolicy`, add the `email_registrable?` method to the `create?` method there, and use a regular `authorize` call? Similar to what we do in `ArticlesController#create`:

```ruby
  def create
    authorize Article
    ...
  end
```

This would completely remove this code from the controller which probably is a good thing.

Reply:
Good thinking! I completely forgot Policy classes are a thing. Updated!